allennlp/training/metrics/perplexity.py

```diff
@@ -0,0 +1,32 @@
+from overrides import overrides
+import torch
+
+from allennlp.training.metrics.average import Average
+from allennlp.training.metrics.metric import Metric
+
+
+@Metric.register("perplexity")
+class Perplexity(Average):
+    """
+    Perplexity is a common metric used for evaluating how well a language model
+    predicts a sample.
+
+    Notes
+    -----
+    Assumes negative log likelihood loss of each batch. Provides the average
```

Comment:
maybe specify that log should be base `e`?

Reply:
```suggestion
    Assumes negative natural log likelihood loss of each batch. Provides the average
```

Reply:
Fixed